internal/system/agent/v2/application/direct/health.go

```diff
@@ -6,33 +6,54 @@
 package direct
 
 import (
-	"fmt"
+	"net/http"
+	"sync"
 
 	"github.com/edgexfoundry/go-mod-registry/v2/registry"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/dtos/common"
 )
 
-const healthy string = "healthy"
+func GetHealth(services []string, rc registry.Client) []interface{} {
+	var mu sync.Mutex
+	var wg sync.WaitGroup
+	var responses []interface{}
 
-func GetHealth(services []string, registryClient registry.Client) map[string]string {
-	health := make(map[string]string)
 	for _, service := range services {
-		if registryClient == nil {
-			health[service] = "registry is required to obtain service health status."
```

Comment:
We need to keep this checking logic to check whether registry client is available for this function, and it should be moved to the beginning of this function.

Reply:
the logic is moved to controller layer

Reply:
this is not protocol specific logic, so it should be located in the application layer.
we might implement other controller protocol in the future, and this logic should be common.